core/ts/Poll.ts

```diff
@@ -404,40 +426,33 @@ export class Poll implements IPoll {
       // e.g if there are 8 messages and the batch size is 5, then
       // the starting index should be 5.
       assert(
-        this.currentMessageBatchIndex === undefined,
-        "The current message batch index should not be defined if this is the first batch",
+        this.currentMessageBatchIndex === this.batchHashes.length - 1,
```

Comment:
when would it be possible that this value is === this.batchHashes.length - 1?

Reply:
resolved